internal/pkg/services/mongodbflex/utils/utils.go

```diff
@@ -157,3 +158,23 @@ func GetUserName(ctx context.Context, apiClient MongoDBFlexClient, projectId, in
 	}
 	return *resp.Item.Username, nil
 }
+
+func GetRestoreStatus(ctx context.Context, apiClient MongoDBFlexClient, projectId, instanceId, backupId string) (string, error) {
+	state := "-"
+	restoreJobs, err := apiClient.ListRestoreJobsExecute(ctx, projectId, instanceId)
+	if err != nil {
+		return "", fmt.Errorf("list restore jobs: %w", err)
+	}
+
+	if restoreJobs.Items == nil {
+		return state, nil
+	}
+
+	for _, restoreJob := range *restoreJobs.Items {
+		if *restoreJob.BackupID == backupId {
+			state = *restoreJob.Status
+			break
+		}
+	}
```

Comment:
What happens if the first restore job in the list (with backupID matching) is not the one being ran now?

Reply:
Not sure if this is something that can happen

Reply:
Not sure if I understood what you meant by "the one being ran now". The method returns the restore status of a specific `backupId`, which is passed as an argument

Reply:
If there are multiple restore jobs, does the API guarantee that the first one in the list is the latest that was ran? That is my question. 
Because if the order in the list is arbitrary, it could happen that this `for loop` finds the first restore job for a backupID that has state `Restored` but in fact the latest restore job that was ran by the user could be further down the list and still have a `In Progress` state.

Reply:
I got it, this is how the Portal is doing it, but nonetheless I will double check with the MongoDBFlex team